src/map/utils/battleutils.cpp

```diff
@@ -2331,99 +2443,213 @@ namespace battleutils
-            {
-                cRatio -= 0.050f * (PDefender->GetMLevel() - PAttacker->GetMLevel());
+
+        //Level correction does not happen in Adoulin zones, Legion, or zones in Escha/Reisenjima
+        //Level correction is only a penalty to players, a player does not get any bonus for fighting lower level monsters
+        //Level correct does give bonuses to Monsters and Avatars. For Avatars it caps at a level difference of 38
+        //I am going to assume that the 38 level difference cap applies to monsters as well
+        //https://www.bluegartr.com/threads/114636-Monster-Avatar-Pet-damage
+        //This thread references the level correction cap for avatars and states that penalties are ignored for avatars
+        //Monster pDIF = Avatar pDIF = Pet pDIF
+        //Based on these points we know monsters and avatars ignore penalties from level correct and only get bonuses so
+        //I believe it is safe to assume all pets do this.
+
+        uint16 zoneId = PAttacker->getZone();
+        //All zones from Adoulin onward have an id of 256+
+        //This includes Escha/Reisenjima and the new Dynamis zones
+        //(Not a post Adoulin Zone) && (Not Legion_A)
+        bool shouldApplyLevelCorrection = (zoneId < 256) && (zoneId != 183);
+
+        ENTITYTYPE attackerType = PAttacker->objtype;
+
+        uint8 attackerLvl = PAttacker->GetMLevel();
+        uint8 defenderLvl = PDefender->GetMLevel();
+        uint8 dLvl = std::abs(attackerLvl - defenderLvl);
+        float correction = ((float)dLvl) * 0.05f;
+
+        //Assuming the cap for mobs is the same as Avatars
+        //Cap at 38 level diff so 38*0.05 = 1.9
+        float cappedCorrection = std::min(correction, 1.9f);
+        
+        if(shouldApplyLevelCorrection)
+        {
+            //Players only get penalties
+            if(attackerType == TYPE_PC)
+            {
+                if(attackerLvl < defenderLvl)
+                {
+                    //Screw the players, no known cap
+                    cRatio -= correction;
+                }
             }
-        }
-        else
-        {
-            if (PAttacker->GetMLevel() > PDefender->GetMLevel())
+            //Mobs, Avatars and pets only get bonuses, no penalties (or they are calculated differently)
+            else if(attackerType == TYPE_MOB || attackerType == TYPE_PET)
             {
-                cRatio += 0.050f * (PAttacker->GetMLevel() - PDefender->GetMLevel());
+                if(attackerLvl > defenderLvl)
+                {
+                    cRatio += cappedCorrection;
+                }
             }
         }
 
-        if (isCritical)
+        float wRatio = cRatio;
+        
+        if(isCritical)
         {
-            cRatio += 1;
+            wRatio += 1;
         }
 
-        cRatio = std::clamp<float>(cRatio, 0, ratioCap);
+        float qRatio = wRatio;
+        float upperLimit = 0.0;
+        float lowerLimit = 0.0;
```

Comment:
Floats should be marked as `0.0f` or else they'll be created as doubles and then converted. I'm sure the compiler probably handles this, but better to be clear